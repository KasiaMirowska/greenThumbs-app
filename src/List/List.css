.list {
    margin-top: 0;
    margin: auto;
    color: rgb(49, 70, 49); 
}
.list-header {

    padding: 20px;

}