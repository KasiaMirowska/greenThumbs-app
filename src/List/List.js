import React from 'react';
import GreenContext from '../Context';
import ListItem from '../LIstItem/ListItem';

export default class List extends React.Component {
    static contextType = GreenContext;
    render() {
       
        let { list } = this.context;
        list = list.map(item => {
            return (
                <ListItem
                    key={item.id}
                    id={item.id}
                    name={item.name}
                    location={item.location}
                    phone={item.display_phone}
                    price={item.price}
                    img={item.image_url}
                    website={item.url}
                    rating={item.rating}
                />
            )
        })

        return (
            <ul>
                <h2>PLACES IN : {this.props.match.params.location}</h2>
                {list}
            </ul>
        )

    }
}