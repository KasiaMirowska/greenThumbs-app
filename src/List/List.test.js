import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as BR} from 'react-router-dom';
import List from './List';

describe('List component', () => {
    const props = {
        match: {
            params: {location: 'location'}
        }
    }
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><List {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><List {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})