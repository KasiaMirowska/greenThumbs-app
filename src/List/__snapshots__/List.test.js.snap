// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<ul
  className="list"
>
  <h2
    className="list-header"
  >
    PLACES IN : 
    LOCATION
  </h2>
</ul>
`;
